### ISSUE TITLE
iOS: Should fail gracefully when JIT is not available

### ISSUE THREAD

**Issue by hrydgard** [hrydgard/ppsspp#14196]
Currently, when JIT is unavailable on iOS (that means, not running from the debugger), the app just crashes and instantly exits. This is not ideal as it's confusing for users what happened. We should probably just automatically switch to IR interpreter instead, and maybe displaying a message.

(A minor challenge when fixing this will be that logging is not available when the debugger isn't attached, AFAIK....)

**Comment by unknownbrackets** [comment]
You can use the web debugger to get logging, although that won't work until after you connect.

-[Unknown]

**Comment by hrydgard** [comment]
Yeah, that does seem to work. Don't know why I keep forgetting that one.

Anyway, it's kinda weird - it seems to happily allow you to mmap and mprotect memory as EXEC, no bad return codes, it just blows up as soon as you try to execute anything. So it seems to be hard to detect, unless I'm missing something of course. Maybe a fork would do it.

I think I'm just gonna have it check for iOS 14.4 or higher, and block attempts to use JIT if the debugger isn't attached (which we do seem to be able to detect).

**Comment by unknownbrackets** [comment]
Is this "resolved" now, given comments in #14240?

-[Unknown]

**Comment by hrydgard** [comment]
I think there are still combinations of states between jailbroken and jit that we don't handle exactly right because we can't detect whether we can execute code or not reliably. although I'm quite thoroughly confused about it all at this point..
